test/built-ins/Object/setPrototypeOf/immutable-prototype-exotic-object-same-value.js

```diff
@@ -0,0 +1,18 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+id: sec-immutable-prototype-exotic-objects-setprototypeof-v
+description: Object.prototype [[SetPrototypeOf]] returns true if value is same
+info: >
+  19.1.3 Properties of the Object Prototype Object
+
+  ...
+  The Object prototype object is an immutable prototype exotic object.
+---*/
+
+var ObjProto = Object.prototype;
+
+assert.sameValue(
+  Object.setPrototypeOf(ObjProto, ObjProto)
```

Comment:
This test case will be valid only if the suggestion on https://github.com/tc39/ecma262/pull/308/files#r50536782 gets merged.